modin/pandas/dataframe.py

```diff
@@ -1335,6 +1335,10 @@ def join(
         """
         Join columns of another ``DataFrame``.
         """
+        if on is not None and not isinstance(other, (Series, DataFrame)):
+            raise ValueError(
+                "Joining multiple DataFrames only supported for joining on index"
```

Comment:
is this error message consistent with what pandas reports?

Reply:
yes